crates/bevy_hierarchy/src/events.rs

```diff
@@ -1,31 +1,19 @@
 use bevy_ecs::{event::Event, prelude::Entity};
 
-/// An [`Event`] that is fired whenever there is a change in the world's hierarchy.
+/// A [`Trigger`] emitted whenever there is a change in the world's hierarchy.
```

Comment:
I'd like to see more clarity in these docs about which entity is triggered. I believe it's the `Parent`, but that should be explicit.

A doc test would also be nice here.

Reply:
Sorry for the long wait! I tried to clarify this a bit now. It's actually triggered for the *child* entity that has the `Parent` component, hence the name `OnParentChange`.

If the event was instead triggered for parents, it would need to be triggered for both the old *and* new parent for the `OnParentChange::Moved` case, which might have a bit more overhead. It could also complicate some logic, since you would need to take into account that a single hierarchy change could trigger the same observer twice (for different entities), and make sure things don't conflict.

Reply:
Also added a simple doc example. Not sure if it should use `World` directly, or have the general `App` setup (`bevy_app` is an optional dependency).

Couldn't make it into an actual doc test nicely, since the observer makes it a bit awkward to test things. I did test it manually though, and the actual tests should cover it.